Sources/Fluent/Query/QueryRepresentable.swift

```diff
@@ -107,6 +107,7 @@ extension QueryRepresentable where Self: ExecutorRepresentable {
 
         if let _ = entity.id, entity.exists {
             // update
+            try type(of: entity).willUpdate(entity: entity)
```

Comment:
Yes that's totally fine with me (though I have a visitor soon so probably tomorrow). However I don't really see how it makes a difference. Since XCode is able to infer which methods are available on `type(of: entity)` I'm guessing this is already a compile-time operation instead of a runtime operation. Is that not correct?